docker/docker_image_dest.go

```diff
@@ -351,8 +359,15 @@ func (d *dockerImageDestination) TryReusingBlob(ctx context.Context, info types.
 				continue
 			}
 		}
-		cache.RecordKnownLocation(d.ref.Transport(), bicTransportScope(d.ref), candidate.Digest, newBICLocationReference(d.ref))
-		return true, types.BlobInfo{Digest: candidate.Digest, Size: size}, nil
+
+		bic.RecordKnownLocation(d.ref.Transport(), bicTransportScope(d.ref), candidate.Digest, newBICLocationReference(d.ref))
+
+		compressionOperation, compressionAlgorithm, err := blobinfocache.OperationAndAlgorithmForCompressor(candidate.CompressorName)
+		if err != nil {
+			return false, types.BlobInfo{}, err
```

Comment:
Yup, making sure we don't overwrite a previously-known compressor value for a given digest makes sense to me.

Reply:
Basically I think this should (log and) `continue`, not fail. See the `Whatever happens here, don't abort the entire operation.` comment.

One way how this could happen is if the cache is updated by some future version to contain a currently-unknown `futureAlgorithm`; then `OperationAndAlgorithmForCompressor` fails.

(BTW we do overwrite a previously-known compressor value, and the BIC tests depend on it due to the `scopeName`+`compressorKnown` loop nesting..)

Reply:
> (BTW we do overwrite a previously-known compressor value

Ah, but `copy.go` does not.

Reply:
Whoops, missed that.  Changing that to log and `continue`, like the `blobExists()`-returned-an-error section.